core/client/src/test/java/alluxio/client/file/FileInStreamTest.java

```diff
@@ -257,6 +257,57 @@ public void testSeek() throws IOException {
   }
 
   /**
+   * Tests seek with incomplete block caching enabled. It seeks within a block forward
+   * and backward, across blocks.
+   */
+  @Test
+  public void seekWithCachingIncompleteBlocksTest() throws IOException {
```

Comment:
one suggestion for the test: IMO, it is more readable / maintainble if we break the big chunk of test case into multiple simpler test cases. For example in your case, I would suggest to have at least four test cases covering {forward,backword}x{withinBlock, crossBlock} four basic scenarios as we discussed offline.

Then based on that, we could have more complicated tests for different seek sequences.

Sorry for pushing on this but I think it is a fairly important and tricky code change. To get better confidence, we should probably have more tests. What do you think?